platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/oauth/OAuthSecurityImpl.java

```diff
@@ -113,6 +108,68 @@ public void setUserTokenOnClient(
     client.header(OAUTH, BEARER + tokenEntry.getAccessToken());
   }
 
+  /**
+   * Gets the user's access token from the token storage to set it to the OAUTH header. If one is
+   * not available, make a call to the OAuth provider to get tokens.
+   *
+   * @param client Non-null client to set the access token on.
+   * @param sourceId the id of the source using OAuth needed to get the correct tokens
+   * @param clientId The client ID registered with the OAuth provider
+   * @param clientSecret The client secret registered with the OAuth provider
+   * @param username The user's username
+   * @param password The user's password
+   * @param discoveryUrl the metadata URL of the OAuth provider
+   */
+  public void setUserTokenOnClient(
```

Comment:
❓ Is the difference between this method and the above `setUserTokenOnClient` method that this method supports retrieving an access token from the Authentication Provider, whereas the above method only supports checking the token storage for an existing access token?

Reply:
Yes, they do different OAuth Flows. The first which just checks the token storage is Authorization Code Flow which is the second portion of the `OAuthPlugin` class. The users have their credentials and are the ones that will provide it to the OAuth provider.
The second is the resource owner password credential flow which means the users give us their credentials and we give it to the OAuth provider.